File: Common/Entity/src/main/java/it/ey/dto/OVPDTO.java
package it.ey.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
public class OVPDTO {
    private Long id;
    private String codice;
    private String descrizione;
    private String contesto;
    private String ambito;
    private String responsabilePolitico;
    private String responsabileAmministrativo;
    private String denominazione;
    private Long valoreIndice;
    private String descrizioneIndice;
    private Long sezione21Id;
    private List<OVPAreaOrganizzativaDTO> areeOrganizzative;
    private List<OVPPrioritaPoliticaDTO> prioritaPolitiche;
    private List<OVPStakeHolderDTO> stakeholders;
    private List<OVPStrategiaDTO> ovpStrategias;
    private List<OVPRisorsaFinanziariaDTO> risorseFinanziarie;

}

File: PIAO_FE/src/app/shared/models/classes/ovp-dto.ts
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { OVPAreaOrganizzativaDTO } from './ovp-area-organizzativa-dto';
import { OVPPrioritaPoliticaDTO } from './ovp-priorita-politica-dto';
import { OVPStakeHolderDTO } from './ovp-stakeholder-dto';
import { OVPStrategiaDTO } from './ovp-strategia-dto';
import { RisorsaFinanziariaDTO } from './risorsa-finanziaria-dto';

export class OVPDTO extends CampiTecniciDTO {
  id?: number;
  denominazione?: string;
  codice?: string;
  descrizione?: string;
  contesto?: string;
  ambito?: string;
  responsabilePolitico?: string;
  responsabileAmministrativo?: string;
  valoreIndice?: number;
  descrizioneIndice?: string;
  sezione21Id?: number;
  areeOrganizzative?: OVPAreaOrganizzativaDTO[];
  prioritaPolitiche?: OVPPrioritaPoliticaDTO[];
  stakeholders?: OVPStakeHolderDTO[];
  ovpStrategias?: OVPStrategiaDTO[];
  risorseFinanziarie?: RisorsaFinanziariaDTO[];
}
